pkgs/by-name/xc/xcp/package.nix

```
@@ -32,6 +33,33 @@ rustPlatform.buildRustPackage (finalAttrs: {
     "test_no_xattr"
     "test_no_perms"
   ];
+  checkFlags =
+    [
+      # had concurrency issues on 64 cores, also tests is quiet fast compared to build
+      "--test-threads=1"
+    ]
+    ++ lib.optionals stdenv.isDarwin [
```

Using `dontUseCargoParallelTests = true` might be cleaner:

```suggestion

  # had concurrency issues on 64 cores, also tests are quiet fast compared to build
  dontUseCargoParallelTests = true;
  checkFlags = lib.optionals stdenv.isDarwin [
```